internal/services/network/virtual_network_gateway_data_source.go

```diff
@@ -326,6 +331,10 @@ func flattenVirtualNetworkGatewayDataSourceIPConfigurations(ipConfigs *[]network
 				}
 			}
 
+			if privateIP := props.PrivateIPAddress; privateIP != nil {
+				v["private_ip_address"] = *privateIP
+			}
```

Comment:
do we want to always set this? ie
```suggestion
			
				v["private_ip_address"] = pointer.From(props.PrivateIPAddress)
			
```

Reply:
Only when private_ip_address_enabled is true, private_ip_address has value. So we may need nil check.

Reply:
Updated